packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -30,6 +30,14 @@ const getImageSize = file => {
     const dimensions = imageSize.sync(
       toArray(fs.readFileSync(file.absolutePath))
     )
+
+    if (!dimensions) {
+      reportError(
+        `gatsby-plugin-sharp couldn't determine dimensions for file:\n${file.absolutePath}\nThis file is unusable and is most likely corrupt.`,
+        ``
```

Comment:
I think we should thread the `reporter` through here. This is likely something we want in structured errors. 

Also, the second argument type is incorrect to pass onto `reporter.error` it should be `typeof undefined | Error` so I would just pass undefined.

Reply:
Good to know, thanks @blainekasten ! Do I just import the reporter directly? I used this reportError function since it happened to be in the file already

Reply:
for now use reportError. Structured erros isn't supported outside of gatsby package.

Reply:
Ugh that sucks, well it does the same as console.error so just use that XD or edit `report-error.js`

Reply:
@wardpeet if I leave it as an empty string it errors normally 🤷‍♂️ I'm hesitant to change `report-error.js` incase any other package is using it for whatever reason.

Reply:
Oh I guess I can just make it not print undefined though

Reply:
Empty string if fine